src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -167,4 +169,114 @@ private static int doSwitch(Object target, int startIndex, Object[] labels) {
         return labels.length;
     }
 
+    /**
+     * Bootstrap method for linking an {@code invokedynamic} call site that
```

Comment:
This should be made clearer - e.g. the first argument is of type `Class` and represents the enum we want to switch on. The remaining constants should be of type `String`, the names of the various constants.

Reply:
That is not quite what the labels represent. The target Enum type is inferred from the bootstrap method's invocation MethodType. (Alternatively, we can add a new Class parameter to the bootstrap method.)

For the labels, please consider this switch:
```
         E sel = null;
         switch (sel) {
             case A -> {}
             case E e && "B".equals(e.name()) -> {}
             case C -> {}
             case E e -> {}
         }
```

The patterns and the constants are mixed, and the order needs to be represented somehow in the labels array, so that the switch will classify the input correctly. The method in this proposal will use `{"A", E.class, "C", E.class}` as the labels array (which is mostly consistent with the `typeSwitch` method), but we could use different encodings if needed.